controllers/grid/settings/sections/form/PKPSectionForm.php

```diff
@@ -138,22 +139,37 @@ public function fetch($request, $template = null, $display = false)
 
     public function initData()
     {
+        $assignedSubeditors = [];
+        $subeditorUserGroups = [];
+
         if ($this->getSection() !== null) {
-            $this->setData([
-                'assignedSubeditors' => Repo::user()
-                    ->getCollector()
-                    ->filterByContextIds([Application::get()->getRequest()->getContext()->getId()])
-                    ->filterByRoleIds($this->assignableRoles)
-                    ->assignedToSectionIds([$this->getSectionId()])
-                    ->getIds()
-                    ->toArray(),
-            ]);
-        } else {
-            $this->setData([
-                'assignedSubeditors' => [],
-            ]);
+            $contextId = Application::get()->getRequest()->getContext()->getId();
+            $assignedSubeditors = Repo::user()
+                ->getCollector()
+                ->filterByContextIds([$contextId])
+                ->filterByRoleIds($this->assignableRoles)
+                ->assignedToSectionIds([$this->getSectionId()])
+                ->getIds()
+                ->toArray();
+
+            if (!empty($assignedSubeditors)) {
+                $subEditorsDao = DAORegistry::getDAO('SubEditorsDAO'); /** @var SubEditorsDAO $subEditorsDao */
+                $subeditorUserGroups = $subEditorsDao->getSectionAssignEditorsUserGroups(
+                    $contextId, 
+                    $this->getSection()->getId(), 
+                    $assignedSubeditors
+                )
+                ->groupBy('user_id')
```

Comment:
This kind of extends DAO operations outside of the DAO (and if I'm reading right, divides some of the database-like operations between SQL in the DAO and collections in memory). Since `getSectionAssignEditorsUserGroups` only exists for this use case, I'd suggest moving the `groupBy` and `map` into the `getSectionAssignEditorsUserGroups` function, and giving it a slightly tighter name, something like `getAssignedUserGroupIds`.

Reply:
Moved to DAO class .